lib/view_component/compiler.rb

```diff
@@ -43,16 +43,29 @@ def compile(raise_errors: false)
       templates.each do |template|
         # Remove existing compiled template methods,
         # as Ruby warns when redefining a method.
-        method_name = call_method_name(template[:variant])
+        pieces = File.basename(template[:path]).split(".")
+
+        method_name =
+          # If the template matches the name of the component,
+          # set the method name with call_method_name
+          if pieces.first == component_class.name.demodulize.underscore
+            call_method_name(template[:variant])
+          # Otherwise, append the name of the template to render_
+          else
+            "render_#{pieces.first.to_sym}"
+          end
 
         if component_class.instance_methods.include?(method_name.to_sym)
           component_class.send(:undef_method, method_name.to_sym)
         end
 
-        component_class.class_eval <<-RUBY, template[:path], -1
-          def #{method_name}
+        component_class.class_eval <<-RUBY, template[:path], -2
+          def #{method_name}(locals = {})
```

Comment:
I wonder if `locals = {}` or `**locals` should be preferred. If the first, we should invoke `ruby2_keywords`

Reply:
I chose the second option